content/docs/error-boundaries.md

```diff
@@ -4,25 +4,25 @@ title: Error Boundaries
 permalink: docs/error-boundaries.html
 ---
 
-In the past, JavaScript errors inside components used to corrupt React’s internal state and cause it to [emit](https://github.com/facebook/react/issues/4026) [cryptic](https://github.com/facebook/react/issues/6895) [errors](https://github.com/facebook/react/issues/8579) on next renders. These errors were always caused by an earlier error in the application code, but React did not provide a way to handle them gracefully in components, and could not recover from them.
+过去，组件内的 JavaScript 错误会导致 React 的内部状态被破坏，并且在下一次渲染时 [产生](https://github.com/facebook/react/issues/4026) [可能无法追踪的](https://github.com/facebook/react/issues/6895) [错误](https://github.com/facebook/react/issues/8579)。这些错误基本上是由其他代码（非React组件代码）中较早的错误引起的，但 React 并没有提供一种在组件中优雅处理这些错误的方式，也无法从错误中恢复。
 
 
-## Introducing Error Boundaries {#introducing-error-boundaries}
+## 错误边界介绍（Error Boundaries） {#introducing-error-boundaries}
 
-A JavaScript error in a part of the UI shouldn’t break the whole app. To solve this problem for React users, React 16 introduces a new concept of an “error boundary”.
+部分 UI 的 JavaScript 错误不应该导致整个应用崩溃，为了解决这个问题，React 16 引入了一个新的概念 —— 错误边界。
 
-Error boundaries are React components that **catch JavaScript errors anywhere in their child component tree, log those errors, and display a fallback UI** instead of the component tree that crashed. Error boundaries catch errors during rendering, in lifecycle methods, and in constructors of the whole tree below them.
+错误边界是 **用于捕获子组件树中任意位置的 JavaScript 错误，记录错误并显示一个备用 UI** 的 React 组件，而不是使整个组件树崩溃。 错误边界在渲染期间、生命周期方法和整个组件树的构造函数中捕获错误。
```

Comment:
```suggestion
错误边界是 **用于捕获子组件树中任意位置的 JavaScript 错误，记录错误并显示一个备用 UI** 的 React 组件。使用错误边界能避免整个组件树因为UI错误而崩溃。 错误边界能在组件渲染期间、组件的生命周期方法和整个组件树的构造函数中捕获错误。
```

这里这个instead of还蛮难翻译的，我倾向于拆开转译一下而不是直接翻译。

Reply:
instead of 可以用 不是....而是.... 供参考

Reply:
觉得这样翻译怎样？
错误边界并不会使整个组件树崩溃，它是 **用于捕获子组件树中任意位置的 JavaScript 错误，记录错误并显示一个备用 UI** 的 React 组件。

Reply:
错误边界是一种 React 组件， 他们不会去捕捉崩溃的组件树，而是*捕捉他们的子组件树中发生的 JavaScript 错误， 记录错误信息，并且显示一个降级的交互界面*